exporter/exporterqueue/queue.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterqueue // import "go.opentelemetry.io/collector/exporter/exporterqueue"
+
+import (
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/internal"
+)
+
+// Queue defines a producer-consumer exchange which can be backed by e.g. the memory-based ring buffer queue
+// (boundedMemoryQueue) or via a disk-based queue (persistentQueue)
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type Queue[T any] internal.Queue[T]
+
+// Settings defines settings for creating a queue.
+type Settings[T any] struct {
+	Sizer            internal.Sizer[T]
+	Capacity         int
+	DataType         component.DataType
+	ExporterSettings exporter.CreateSettings
+}
+
+type Factory[T any] func(Settings[T]) Queue[T]
+
+// NewMemoryQueueFactory returns a factory to create a new memory queue.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+func NewMemoryQueueFactory[T any]() Factory[T] {
+	return func(set Settings[T]) Queue[T] {
```

Comment:
Should accept Context, Settings and Config. See pattern here https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/extensions/extensions.go#L174

Where Settings are things like TelemetrySettings, etc. In your case Sizer and Capacity should be passed and calculated from Config.

Reply:
done